lib/dockerregistry/checks.go

```
@@ -0,0 +1,133 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inventory
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	gogit "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+// ImageRemovalCheck implements the PreCheck interface and checks against
+// pull requests that attempt to remove any images from the promoter manifests.
+type ImageRemovalCheck struct {
+	GitRepoURL     string
+	MasterSHA      plumbing.Hash
+	PullRequestSHA plumbing.Hash
+}
+
+// MKRealImageRemovalCheck returns an instance of ImageRemovalCheck.
+func MKRealImageRemovalCheck(gitRepoURL string) *ImageRemovalCheck {
+	masterSHAStr := os.Getenv("PULL_BASE_SHA")
+	pullRequestSHAStr := os.Getenv("PULL_PULL_SHA")
```

What happens if this is an empty var? Fail early if possible...